### DISCUSSION TITLE
Webpack: SASS issue/ NPM not starting

### DISCUSSION THREAD

**Discussion by thedigitalmenagerie** [nss-evening-cohort-14/evening-client-side#98] [Help Me!] [answered]
The .env file is not pulling down to my local when cloning. I tried to git clone several times and each time it would tell me that i could fetch and push after checking git remote -v.

I tried adding and committing it to master and then proceeding to do npm i, this gives errors that instruct me to npm fund and npm audit fix --force. This will start the browser but it only says 'test' with no other messages.

https://www.npmjs.com/package/dotenv-webpack
https://stackoverflow.com/questions/38983153/git-ignore-env-files-not-working
https://www.xspdf.com/resolution/50378798.html

https://github.com/thedigitalmenagerie/sorting-hat-webpack

<img width="1440" alt="Screen Shot 2021-02-15 at 8 12 38 PM" src="https://user-images.githubusercontent.com/76716670/108010331-3d73ac80-6fca-11eb-90cc-34bac561873a.png">

**Comment by drteresavasquez** [comment]
@thedigitalmenagerie I see the `.env` file in your list of files. I think there may be something else happening here. Let's jump in Hackery 1 to take a look.

**Comment by thedigitalmenagerie** [comment]
I accidentally cloned the webpack-template from Dr. T's personal page as opposed to the NSS webpack-template.

**Reply by drteresavasquez** [reply]
Easy mistake! Glad you found the solution! Let us know if you have any other questions.

**Comment by thedigitalmenagerie** [comment]
Looks like command 'npm i' is still throwing an error code 1. I found a stackoverflow article that said run 'npm install node-sass'. It still gives the 'npm fund' message but no errors.

https://stackoverflow.com/questions/47537309/an-error-occurred-while-running-npm-i-exit-code-1

Command 'npm start", however keeps giving me the following error:

<img width="1440" alt="Screen Shot 2021-02-15 at 8 41 56 PM" src="https://user-images.githubusercontent.com/76716670/108012664-8e39d400-6fcf-11eb-9e2f-0ad70fc35819.png">

I tried four suggestions on this page and none of them seem to be working. 


https://stackoverflow.com/questions/46419024/error-in-node-modules-css-loader

**Comment by thedigitalmenagerie** [comment]
Ok. There were quite a few problems here. After Dr.T helped me identify the incorrect repo and the fact that Xcode Command Line Tools were not installed, I was still unable to make 'npm i 'or 'npm start' work, due to my pre-owned laptop already being updated to Big Sur.

I finally found the answer here:

https://stackoverflow.com/questions/64612707/node-sass-does-not-yet-support-your-current-environment-windows-64-bit-with-uns

Command line 'npm i gulp-sass --save-dev' worked and I displayed to the page!

<img width="1440" alt="Screen Shot 2021-02-15 at 11 31 14 PM" src="https://user-images.githubusercontent.com/76716670/108022563-64d87280-6fe6-11eb-810a-e5e08bc412ff.png">

**Reply by drteresavasquez** [reply]
Let's have another look tomorrow. We want to make sure you're using the right version of sass so you may need to downgrade your node version. Run `node -v` in your terminal and tell me what it gives you.

**Reply by thedigitalmenagerie** [reply]
v15.5.1

**Reply by drteresavasquez** [reply]
That's the issue. We need to downgrade your node. We can walk through it in class but I think someone in E13 had a similar problem. Let me see if I can find that ticket.

**Comment by thedigitalmenagerie** [comment, marked as answer]
Instead of using gulp sass, it was necessary that I downgrade my node version to 14. Now, it is up and running in a way that is compatible with webpack!
